runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/config/MqttOptionsConfigAdapter.java

```diff
@@ -166,7 +167,11 @@ public OptionsConfig adaptFromJson(
         if (object.containsKey(VERSIONS_NAME))
         {
             List<Integer> versions = object.getJsonArray(VERSIONS_NAME).stream()
-                .map(item -> ((JsonNumber) item).intValue())
+                .map(item ->
+                {
+                    String s = ((JsonString) item).getString();
+                    return s.equals(MQTT_VERSION_4) ? 4 : 5;
+                })
```

Comment:
With the `MqttVersion` enum, this can use a static method on `MqttVersion` to lookup by version string.